lib/batch/batch_metadata_updates_634.rb

```diff
@@ -0,0 +1,73 @@
+# batch updates for https://github.com/galterlibrary/digital-repository/issues/634
+
+# Updates number 1:
+#   - All items with "Pediatric Neurology Briefs Publishers" in the Original 
+#   - Publisher field --> Add additional field to Digital Publisher with "Pediatric 
+#   - Neurology Briefs Publishers" and delete from Original Publisher
+
+pnb_publishers = "Pediatric Neurology Briefs Publishers"
+items_needing_original_publisher_updates = GenericFile.where("original_publisher" => pnb_publishers) + 
+                                           Collection.where("original_publisher" => pnb_publishers)
+
+items_needing_original_publisher_updates.each do |item|
+  item.publisher += [pnb_publishers]
+  item.original_publisher -= [pnb_publishers]
+  item.save
+end
+
+
+# Updates number 2:
+#   - All items in Science in Society Collection - delete "Science in Society" 
+#   - from Original Publisher 
+
+sis_publisher = "Science in Society"
+
+def update_sis_original_publisher(id)
+  collection = Collection.find(id)
+
+  if !collection.members.empty?
+    collection.members.each do |member|
+      if member.class == Collection 
+        update_sis_original_publisher(member.id)
+      else
+        member.original_publisher -= [sis_publisher]
+        member.save
+      end
+    end
+  end
+
+  collection.original_publisher -= [sis_publisher]
+  collection.save!
+end
+
+sis_collection = "93c81706-47f8-49a1-88a2-3acbc971f4ed"
+update_sis_original_publisher(sis_collection)
+
+
+# Updates number 3:
+#   - All DigitalHub items - delete "Chicago, Illinois, United States" 
+#   - and delete "Chicago, IL, USA" from Location
+
+chicago_locations = ["Chicago, Illinois, United States", "Chicago, IL, USA"]
+items_needing_based_near_updates = GenericFile.where("based_near" => chicago_locations) +
+                                   Collection.where("based_near" => chicago_locations)
+
+items_needing_based_near_updates.each do |item| 
+  item.based_near -= chicago_locations
+  item.save
+end
+
+
+# Updates number 4:
+#   - All DigitalHub items with Digital Publisher value 
+#   - "Galter Health Sciences Library, Feinberg School of Medicine, 
+#   - Northwestern University" --> change to "Galter Health Sciences Library"
+
+ghsl_fsm_nu_publisher = ["Galter Health Sciences Library, Feinberg School of Medicine, Northwestern University"]
+items_needing_publisher_updates = GenericFile.where("publisher" => ghsl_fsm_nu_publisher) + 
+                                  Collection.where("publisher" => ghsl_fsm_nu_publisher)
+
+items_needing_publisher_updates.each do |item|
+  item.publisher -= ghsl_fsm_nu_publisher
+  item.save
+end
```

Comment:
You were saying you had issues with the script not updating records. How did you test that this works?

Reply:
So, `item.publisher` is an array of values. For some reason, using the shovel operator, `<<` was invalid. Using the `-=` or `+=` operators returns a new instance of the array, and was valid in saving.

Reply:
But I should refactor this to `save!` to rescue instead of just returning false.

Reply:
I was also asking if you had forgotten to commit the tests you made for this? Or maybe you ran this on your local db?

Reply:
There already is a test that checks updating these values. It took me time to realize that none of these tests were using the `<<` operator, and that it appeared to be in a specific format that was valid for updating. That was when I tried the specific way of updating, which is using `-=` , `+=`, or just `=`.